### DISCUSSION TITLE
Spotlight search

### DISCUSSION THREAD

**Discussion by Toemsel** [godotengine/godot-proposals#10760] [Editor]
Having a ton of resources, scripts, scenes,... the FileSystem explorer's search function is cumbersome. You always have to search, delete, search, delete,... modern IDEs (VS Code with ctrl+p, VS with ctrl+,) offer a spotlight search. You enter a search term and receive direct hits or elements with a certain distance match. This would improve productivity and you don't have to search in the FileSystem, Script and Methods tab all the time.

The spotlight search would pop up mid-screen and offer results as you type. The search should include filenames, contents of shaders, and gdscript.

**Comment by sockeye-d** [comment]
There's already a quick open dialog (I think the shortcut is ~~Ctrl+alt+P~~ shift+alt+o) where you can filter resources, scenes, and scripts based on their file name.
It could be extended to show a preview for scripts/shaders and also to search within the files and, for scripts, the script symbols (functions and stuff)

**Reply by Toemsel** [reply]
> There's already a quick open dialog (I think the shortcut is Ctrl+alt+P) where you can filter resources, scenes, and scripts based on their file name. It could be extended to show a preview for scripts/shaders and also to search within the files and, for scripts, the script symbols (functions and stuff)

Does not include the file's content nor can I use that search function without a mouse. There is no shortcut to navigate around within the treeview after typing.

**Reply by sockeye-d** [reply]
Oops, it's shift+alt+o. I got the shortcut wrong, sorry

**Reply by Toemsel** [reply]
> Oops, it's shift+alt+o. I got the shortcut wrong, sorry

No problem pal. That is helpful - thanks. Still not what I am looking for, but it is better than nothing ;)

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/3788.
